Source/WebCore/page/SecurityOriginData.h

```diff
@@ -42,19 +49,33 @@ struct SecurityOriginData {
     {
         RELEASE_ASSERT_WITH_SECURITY_IMPLICATION(!isHashTableDeletedValue());
     }
+    explicit SecurityOriginData(OpaqueOriginIdentifier opaqueOriginIdentifier)
+        : protocol(emptyString())
+        , host(emptyString())
+        , port(std::nullopt)
+        , opaqueOriginIdentifier(opaqueOriginIdentifier)
+    {
+    }
+    // This constructor should only be used in cases where the origin must not be opaque. SecurityOriginData::fromURL should be used in most cases.
+    explicit SecurityOriginData(const URL& url)
+        : SecurityOriginData {
+            url.protocol().isNull() ? emptyString() : url.protocol().convertToASCIILowercase()
+            , url.host().isNull() ? emptyString() : url.host().convertToASCIILowercase()
+            , url.port()
+        }
+    {
+    }
```

Comment:
Good point, this constructor should create an opaque origin if the URL is null. I'll fix that. And regarding constructor verses `fromURL()`, I initially added the flag to `fromURL()` but I realized creating a constructor seemed more natural in this case, so I change it. But, choosing the correct method relies on some subtlety, so I'll go back to using the flag since that seems better and more explicit.